src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,794 @@
+        )
+
+        return cmd
+
+    def get_verilog_top_module_intf_names(self):
+        """Return a dict of names of input and output interfaces.
+        The keys reflect the protocols each interface implements:
+        'clk', 'rst', 'm_axis', 's_axis', 'aximm', 'axilite'.
+        Values are lists of tuples (axis, aximm) or names (axilite):
+        'axis' tuples correspond to the list of node inputs in order,
+        each tuple is (interface_name, interface_width_bits).
+        axilite always assumed to be 32 bits and is not tuple (name only).
+        Each block must have at most one aximm and one axilite."""
+
+        intf_names = {}
+        intf_names["clk"] = ["ap_clk"]
+        intf_names["rst"] = ["ap_rst_n"]
+        intf_names["s_axis"] = [("in0_V", self.get_instream_width_padded())]
+        intf_names["m_axis"] = [("out_V", self.get_outstream_width_padded())]
+        intf_names["aximm"] = []
+        intf_names["axilite"] = []
+        intf_names["ap_none"] = []
+        if self.get_nodeattr("runtime_writeable_weights") == 1:
+            intf_names["axilite"] = ["s_axilite"]
+
+        return intf_names
+
+    def get_dynamic_config(self, weights, address_stride=1):
+        """Returns a configuration dictionary containing axilite write commands
+        in order to program the thresholds into the RTL core during runtime.
+        The default address stride for the weights is 1 byte."""
+
+        # thresholds = model.get_initializer(self.onnx_node.input[1])
+        thresholds = weights
+        num_channels, num_weights_per_channel = thresholds.shape
+
+        weight_addr_boundary = find_next_power_of_2(num_weights_per_channel)
+        # Make sure that the next power of 2 (output) is greater than the input
+        assert weight_addr_boundary >= num_weights_per_channel
+
+        config = {}
+        channel_cntr = 0
+        wdt = self.get_weight_datatype()
+        bw_hexdigit = roundup_to_integer_multiple(wdt.bitwidth(), 4)
+        for channel in thresholds:
+            channel_start_addr = channel_cntr * weight_addr_boundary * address_stride
+            weight_cntr = 0
+            addr = 0
+            for weight in channel:
+                key_name = "{}_{}{}_{}{}".format(
+                    "axilite", "ch", str(channel_cntr), "w", str(weight_cntr)
+                )
+                config[key_name] = (
+                    channel_start_addr + addr,
+                    int(
+                        str(
+                            pack_innermost_dim_as_hex_string(
+                                [weight],
+                                wdt,
+                                bw_hexdigit,
+                            )
+                        ),
+                        0,
+                    ),
+                )
+
+                weight_cntr += 1
+                addr += address_stride
+
+            channel_cntr += 1
+
+        return config
+
+    def generate_params(self, model, path):
```

Comment:
This function seems does not seem to be used for the RTL variant and can be removed.